### PR TITLE
Gradle Manager

### PR DESCRIPTION
**Pull request by matthewchivers**
Initial interfaces / work towards the Gradle Manager.

This manager will enable the installation and usage of Gradle on target images.

### CODE REVIEW

File: galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.gradle.manager/bnd.bnd [hunk 1]
```diff
@@ -0,0 +1,6 @@
+-snapshot: ${tstamp}
```
Comment (Michael-Baylis) [inline galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.gradle.manager/bnd.bnd:1]: best placed in the languages parent,  not the testingtools parent

File: galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.gradle.ubuntu.manager/src/main/java/dev/galasa/gradle/ubuntu/GradleUbuntuInstallation.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    /**
+     * The tag to be assigned to this Gradle installation.
+     */
+    String gradleTag() default "PRIMARY";
+
+    /**
+     * The tag of the Linux Image this installation is to be associated with.
+     */
+    String imageTag() default "PRIMARY";
```
Comment (Michael-Baylis) [inline galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.gradle.ubuntu.manager/src/main/java/dev/galasa/gradle/ubuntu/GradleUbuntuInstallation.java:43]: dont we need to tie this with a java tag as well,  so we can generate the correct gradle command etc

**Review by Michael-Baylis — CHANGES_REQUESTED** [review]

File: galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.gradle.ubuntu.manager/src/main/java/dev/galasa/gradle/ubuntu/GradleUbuntuInstallation.java (hunk 2 above)
Comment (matthewchivers) [reply galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.gradle.ubuntu.manager/src/main/java/dev/galasa/gradle/ubuntu/GradleUbuntuInstallation.java:43]: Yes, you're quite right. I know the Gradle client uses the java set in $JAVA_HOME, but yes, there is the Gradle property `org.gradle.java.home` that can be set for Gradle to use for the build process. I'll add the tag with the idea to use it when setting this property for Gradle.

File: galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.gradle.manager/bnd.bnd (hunk 1 above)
Comment (matthewchivers) [reply galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.gradle.manager/bnd.bnd:1]: Will fix.

File: galasa-managers-parent/galasa-managers-languages-parent/dev.galasa.gradle.ubuntu.manager/build.gradle [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ * Copyright contributors to the Galasa project 
+ */
+
+plugins {
+    id 'galasa.manager'
+}
+
+description = 'Galasa Gradle Ubuntu Manager'
+
+version = '0.18.0'
```
Comment (Michael-Baylis) [inline galasa-managers-parent/galasa-managers-languages-parent/dev.galasa.gradle.ubuntu.manager/build.gradle:11]: 19 now

**Review by Michael-Baylis — CHANGES_REQUESTED** [review]
minor changes please

**Pull request closed without merging**